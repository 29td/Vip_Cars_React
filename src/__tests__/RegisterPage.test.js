import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/';
import { BrowserRouter as Router } from 'react-router-dom';
import { AuthProvider } from 'react-auth-kit';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import RegistrationPage from '../components/RegisterPage';

describe('Login component', () => {
  test('Should render the proper Register component', () => {
    const register = render(
      <React.StrictMode>
        <Provider store={store}>
          <AuthProvider
            authType="cokie"
            authName="_auth"
            cookieDomain={window.location.hostname}
            cookieSecure={false}
          >
            <Router>
              <RegistrationPage />
            </Router>
          </AuthProvider>
        </Provider>
        ,
      </React.StrictMode>,
    );
    expect(register).toMatchSnapshot();
  });
  test('Should have Log In link', () => {
    render(
      <React.StrictMode>
        <Provider store={store}>
          <AuthProvider
            authType="cokie"
            authName="_auth"
            cookieDomain={window.location.hostname}
            cookieSecure={false}
          >
            <Router>
              <RegistrationPage />
            </Router>
          </AuthProvider>
        </Provider>
        ,
      </React.StrictMode>,
    );
    const line = document.querySelector('.line');
    expect(line).toHaveTextContent('Log In');
  });
  test('Should Sign Up Button', () => {
    render(
      <React.StrictMode>
        <Provider store={store}>
          <AuthProvider
            authType="cokie"
            authName="_auth"
            cookieDomain={window.location.hostname}
            cookieSecure={false}
          >
            <Router>
              <RegistrationPage />
            </Router>
          </AuthProvider>
        </Provider>
        ,
      </React.StrictMode>,
    );
    const signUp = document.querySelector('button');
    expect(signUp).toHaveTextContent('Sign Up');
  });
  test('Should have Register title', () => {
    render(
      <React.StrictMode>
        <Provider store={store}>
          <AuthProvider
            authType="cokie"
            authName="_auth"
            cookieDomain={window.location.hostname}
            cookieSecure={false}
          >
            <Router>
              <RegistrationPage />
            </Router>
          </AuthProvider>
        </Provider>
        ,
      </React.StrictMode>,
    );
    const title = document.querySelector('h1');
    expect(title).toHaveTextContent('Register');
  });
});
