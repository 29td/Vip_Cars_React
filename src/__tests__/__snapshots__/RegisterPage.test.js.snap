// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Should render the proper Register component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="session"
      >
        <section>
          <h1>
            Register
          </h1>
          <p
            aria-live="assertive"
            class="offscreen"
          />
          <form>
            <label
              for="username"
            >
              Username:
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-check hide"
                data-icon="check"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
                  fill="currentColor"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-xmark hide"
                data-icon="xmark"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
                  fill="currentColor"
                />
              </svg>
              <br />
              <input
                aria-describedby="uidnote"
                aria-invalid="true"
                autocomplete="off"
                class=""
                id="username"
                placeholder="Username"
                required=""
                title="User name"
                type="text"
              />
            </label>
            <p
              class="offscreen"
              id="uidnote"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle-info "
                data-icon="circle-info"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-144c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
                  fill="currentColor"
                />
              </svg>
              3 to 24 characters.
              <br />
              Must begin with a letter.
              <br />
              Letters, numbers, underscores, hyphens allowed.
            </p>
            <label
              for="email"
            >
              Email:
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-check hide"
                data-icon="check"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
                  fill="currentColor"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-xmark hide"
                data-icon="xmark"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
                  fill="currentColor"
                />
              </svg>
              <br />
              <input
                aria-describedby="emailnote"
                aria-invalid="true"
                autocomplete="off"
                class=""
                id="email"
                placeholder="Email"
                required=""
                title="Email"
                type="email"
              />
            </label>
            <p
              class="offscreen"
              id="emailnote"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle-info "
                data-icon="circle-info"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-144c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
                  fill="currentColor"
                />
              </svg>
              5 or more characters.
              <br />
              Must have the
               
              <span
                aria-label="at symbol"
              >
                @
              </span>
               
              letter.
              <br />
              Must have a top-level domain such as ‘.com’
            </p>
            <label
              for="password"
            >
              Password:
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-check hide"
                data-icon="check"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
                  fill="currentColor"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-xmark hide"
                data-icon="xmark"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
                  fill="currentColor"
                />
              </svg>
              <br />
              <input
                aria-describedby="pwdnote"
                aria-invalid="true"
                class=""
                id="password"
                placeholder="Password"
                required=""
                title="Password"
                type="password"
                value=""
              />
            </label>
            <p
              class="offscreen"
              id="pwdnote"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle-info "
                data-icon="circle-info"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-144c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
                  fill="currentColor"
                />
              </svg>
              8 to 24 characters.
              <br />
              Allowed special characters:
               
              <span
                aria-label="exclamation mark"
              >
                !
              </span>
               
              <span
                aria-label="at symbol"
              >
                @
              </span>
               
              <span
                aria-label="hashtag"
              >
                #
              </span>
               
              <span
                aria-label="dollar sign"
              >
                $
              </span>
               
              <span
                aria-label="percent"
              >
                %
              </span>
            </p>
            <label
              for="confirm_pwd"
            >
              Confirm Password:
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-check hide"
                data-icon="check"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
                  fill="currentColor"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-xmark hide"
                data-icon="xmark"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
                  fill="currentColor"
                />
              </svg>
              <br />
              <input
                aria-describedby="confirmnote"
                aria-invalid="false"
                class=""
                id="confirm_pwd"
                placeholder="Confirm Password"
                required=""
                title="Confirm Password"
                type="password"
                value=""
              />
            </label>
            <p
              class="offscreen"
              id="confirmnote"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle-info "
                data-icon="circle-info"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-144c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
                  fill="currentColor"
                />
              </svg>
              Must match the first password input field.
            </p>
            <br />
            <button
              disabled=""
              type="submit"
            >
              Sign Up
            </button>
          </form>
          <p>
            Already registered?
             
            <span
              class="line"
            >
              <a
                href="/login"
              >
                Log In
              </a>
            </span>
          </p>
        </section>
      </main>
      ,
    </div>
  </body>,
  "container": <div>
    <main
      class="session"
    >
      <section>
        <h1>
          Register
        </h1>
        <p
          aria-live="assertive"
          class="offscreen"
        />
        <form>
          <label
            for="username"
          >
            Username:
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-check hide"
              data-icon="check"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark hide"
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
                fill="currentColor"
              />
            </svg>
            <br />
            <input
              aria-describedby="uidnote"
              aria-invalid="true"
              autocomplete="off"
              class=""
              id="username"
              placeholder="Username"
              required=""
              title="User name"
              type="text"
            />
          </label>
          <p
            class="offscreen"
            id="uidnote"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-circle-info "
              data-icon="circle-info"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-144c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
                fill="currentColor"
              />
            </svg>
            3 to 24 characters.
            <br />
            Must begin with a letter.
            <br />
            Letters, numbers, underscores, hyphens allowed.
          </p>
          <label
            for="email"
          >
            Email:
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-check hide"
              data-icon="check"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark hide"
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
                fill="currentColor"
              />
            </svg>
            <br />
            <input
              aria-describedby="emailnote"
              aria-invalid="true"
              autocomplete="off"
              class=""
              id="email"
              placeholder="Email"
              required=""
              title="Email"
              type="email"
            />
          </label>
          <p
            class="offscreen"
            id="emailnote"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-circle-info "
              data-icon="circle-info"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-144c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
                fill="currentColor"
              />
            </svg>
            5 or more characters.
            <br />
            Must have the
             
            <span
              aria-label="at symbol"
            >
              @
            </span>
             
            letter.
            <br />
            Must have a top-level domain such as ‘.com’
          </p>
          <label
            for="password"
          >
            Password:
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-check hide"
              data-icon="check"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark hide"
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
                fill="currentColor"
              />
            </svg>
            <br />
            <input
              aria-describedby="pwdnote"
              aria-invalid="true"
              class=""
              id="password"
              placeholder="Password"
              required=""
              title="Password"
              type="password"
              value=""
            />
          </label>
          <p
            class="offscreen"
            id="pwdnote"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-circle-info "
              data-icon="circle-info"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-144c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
                fill="currentColor"
              />
            </svg>
            8 to 24 characters.
            <br />
            Allowed special characters:
             
            <span
              aria-label="exclamation mark"
            >
              !
            </span>
             
            <span
              aria-label="at symbol"
            >
              @
            </span>
             
            <span
              aria-label="hashtag"
            >
              #
            </span>
             
            <span
              aria-label="dollar sign"
            >
              $
            </span>
             
            <span
              aria-label="percent"
            >
              %
            </span>
          </p>
          <label
            for="confirm_pwd"
          >
            Confirm Password:
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-check hide"
              data-icon="check"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark hide"
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
                fill="currentColor"
              />
            </svg>
            <br />
            <input
              aria-describedby="confirmnote"
              aria-invalid="false"
              class=""
              id="confirm_pwd"
              placeholder="Confirm Password"
              required=""
              title="Confirm Password"
              type="password"
              value=""
            />
          </label>
          <p
            class="offscreen"
            id="confirmnote"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-circle-info "
              data-icon="circle-info"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-144c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
                fill="currentColor"
              />
            </svg>
            Must match the first password input field.
          </p>
          <br />
          <button
            disabled=""
            type="submit"
          >
            Sign Up
          </button>
        </form>
        <p>
          Already registered?
           
          <span
            class="line"
          >
            <a
              href="/login"
            >
              Log In
            </a>
          </span>
        </p>
      </section>
    </main>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
